File: diffai-python/test_simple.py
#!/usr/bin/env python
"""Simple test for diffai_python Rust integration"""

try:
    import diffai_python
    print("✅ Successfully imported diffai_python")
    print(f"Module path: {diffai_python.__file__}")
    print(f"Available functions: {dir(diffai_python)}")
    print(f"Version: {diffai_python.__version__}")
    
    # Test basic diff functionality (using diff_py as per diffx pattern)
    old_data = {"name": "test", "value": 123}
    new_data = {"name": "test", "value": 456}
    
    results = diffai_python.diff_py(old_data, new_data)
    print(f"✅ Diff results: {results}")
    print("🎉 diffai_python Rust integration working correctly!")
    
except ImportError as e:
    print(f"❌ Import failed: {e}")
except Exception as e:
    print(f"❌ Test failed: {e}")

File: diffai-js/test_simple.js
#!/usr/bin/env node
/**
 * Simple test for diffai-js Node.js integration
 */

try {
    const diffai = require('./index.js');
    console.log('✅ Successfully imported diffai-js');
    console.log(`Available functions:`, Object.keys(diffai));
    
    // Test basic diff functionality
    const oldData = { name: "test", value: 123 };
    const newData = { name: "test", value: 456 };
    
    const results = diffai.diff(oldData, newData);
    console.log(`✅ Diff results:`, results);
    
    // Test AI/ML specific functionality
    const oldModel = {
        learning_rate: 0.001,
        accuracy: 0.85,
        model_type: "pytorch"
    };
    
    const newModel = {
        learning_rate: 0.01,
        accuracy: 0.92,
        model_type: "pytorch"
    };
    
    const mlResults = diffai.diff(oldModel, newModel, {
        learningRateTracking: true,
        accuracyTracking: true
    });
    console.log(`✅ ML analysis results:`, mlResults);
    
    console.log('🎉 diffai-js Node.js integration working correctly!');
    
} catch (error) {
    console.error(`❌ Test failed:`, error.message);
    process.exit(1);
}